docs/opup.rst

```diff
@@ -0,0 +1,85 @@
+.. _opup:
+
+OpUp:  Budget Increase Utility
+========================
+
+Some opcode budget is consumed during execution of every Algorand Smart Contract because every teal
+instruction has a corresponding cost. In order for the evaluation to succeed, the budget consumed must not
+exceed the budget provided. This constraint may introduce a problem for expensive contracts that quickly
+consume the initial budget of 700. The OpUp budget increase utility provides a workaround using inner
+transactions. Note that there is a context specific limit to the number of inner transactions issued in a
+transaction group so budget cannot be increased arbitrarily. The available budget when using the OpUp
+utility will need to be high enough to execute the teal code that issues the inner transactions. A budget
+of ~20 is enough for most use cases.
```

Comment:
nit: it would be nice if the overview explicitly stated that how this works (by executing noop inner txns that increase the group's pooled compute budget) and where the funding comes from (the app's account, not the caller or creator's). The first is a nice-to-have, but the second seems more important for practical purposes